src/Microsoft.Health.Fhir.SqlServer/Features/Storage/Registry/SqlServerSearchParameterStatusDataStore.cs

```diff
@@ -155,15 +161,59 @@ public async Task UpsertStatuses(IReadOnlyCollection<ResourceSearchParameterStat
             // If the search parameter table in SQL does not yet contain the larger status column we reset back to disabled status
             if (_schemaInformation.Current < (int)SchemaVersion.V52)
             {
-                foreach (var status in statuses)
+                foreach (var status in statuses.Where(s => s.Status == SearchParameterStatus.Unsupported))
                 {
-                    if (status.Status == SearchParameterStatus.Unsupported)
+                    status.Status = SearchParameterStatus.Disabled;
+                }
+            }
+
+            await UpsertStatusesWithRetry(statuses, 3, cancellationToken);
+        }
+
+        private async Task UpsertStatusesWithRetry(IReadOnlyCollection<ResourceSearchParameterStatus> statuses, int maxRetries, CancellationToken cancellationToken)
+        {
+            var currentStatuses = statuses.ToList();
+            int retryCount = 0;
+
+            while (retryCount <= maxRetries)
+            {
+                try
+                {
+                    await UpsertStatusesInternal(currentStatuses, cancellationToken);
+                    return; // Success
+                }
+                catch (SqlException sqlEx) when (sqlEx.Number == 50001 && retryCount < maxRetries) // Our custom concurrency error
+                {
+                    // Optimistic concurrency conflict detected - refresh and retry
+                    retryCount++;
+                    _logger.LogWarning("Optimistic concurrency conflict detected on attempt {RetryCount}. Retrying...", retryCount);
+
+                    // Refresh the statuses with current LastUpdated values
+                    var refreshedStatuses = await GetSearchParameterStatuses(cancellationToken);
+                    var refreshedDict = refreshedStatuses.ToDictionary(s => s.Uri.OriginalString, s => s);
+
+                    // Update our statuses with fresh LastUpdated values
+                    foreach (var status in currentStatuses)
                     {
-                        status.Status = SearchParameterStatus.Disabled;
+                        if (refreshedDict.TryGetValue(status.Uri.OriginalString, out var refreshed))
+                        {
+                            status.LastUpdated = refreshed.LastUpdated;
```

Comment:
This is having a conflict returned from SQL saying that the version we have is out of date as shown by the last updated time not matching. This then just overrides the last updated time locally to match the DB and tries again. What does this gain over just not checking the last updated at all? It looks like we are just ignoring any other change in the DB, we aren't trying to resolve the conflict with the DB.